src/rattler_build_conda_compat/variant_config.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from itertools import product
+
+
+def variant_combinations(data: dict[str, str | list[str]]) -> list[dict[str, str]]:
```

Comment:
Could definitely use a function docstring and explain what the inputs and outputs are.